app/src/main/java/com/kickstarter/viewmodels/ProjectHolderViewModel.java

```diff
@@ -37,141 +43,243 @@
+     */
     Observable<Integer> projectStateViewGroupBackgroundColorInt();
 
-    /** Emits when the project state view group should be gone. */
+    /**
+     * Emits when the project state view group should be gone.
+     */
     Observable<Boolean> projectStateViewGroupIsGone();
 
-    /** Emits when we should set default stats margins. */
+    /**
+     * Emits when we should set default stats margins.
+     */
     Observable<Boolean> shouldSetDefaultStatsMargins();
 
-    /** Emits when we should set the canceled state view. */
+    /**
+     * Emits when we should set the canceled state view.
+     */
     Observable<Void> setCanceledProjectStateView();
 
-    /** Emits when we should set an on click listener to the social view. */
+    /**
+     * Emits when we should set an on click listener to the social view.
+     */
     Observable<Void> setProjectSocialClickListener();
 
-    /** Emits when we should set the successful state view. */
+    /**
+     * Emits when we should set the successful state view.
+     */
     Observable<DateTime> setSuccessfulProjectStateView();
 
-    /** Emits when we should set the suspended state view. */
+    /**
+     * Emits when we should set the suspended state view.
+     */
     Observable<Void> setSuspendedProjectStateView();
 
-    /** Emits when we should set the unsuccessful state view. */
+    /**
+     * Emits when we should set the unsuccessful state view.
+     */
     Observable<DateTime> setUnsuccessfulProjectStateView();
 
-    /** Emits when we should start the {@link com.kickstarter.ui.activities.ProjectSocialActivity}. */
+    /**
+     * Emits when we should start the {@link com.kickstarter.ui.activities.ProjectSocialActivity}.
+     */
     Observable<Project> startProjectSocialActivity();
 
-    /** Emits the updates count for display. s*/
+    /**
+     * Emits the updates count for display. s
+     */
     Observable<String> updatesCountTextViewText();
 
-    /** Emits the usd conversion text for display. */
+    /**
+     * Emits the usd conversion text for display.
+     */
     Observable<Pair<String, String>> usdConversionPledgedAndGoalText();
 
-    /** Emits when the usd conversion view should be gone. */
+    /**
+     * Emits when the usd conversion view should be gone.
+     */
     Observable<Boolean> usdConversionTextViewIsGone();
   }
 
   final class ViewModel extends ActivityViewModel<ProjectViewHolder> implements Inputs, Outputs {
+    private final ApiClientType client;
     private final KSCurrency ksCurrency;
+    private final UserCurrency userCurrency;
+    private final CurrentUserType currentUser;
 
     public ViewModel(final @NonNull Environment environment) {
       super(environment);
+
+      this.client = environment.apiClient();
+      this.currentUser = environment.currentUser();
       this.ksCurrency = environment.ksCurrency();
+      this.userCurrency = environment.userCurrency();
+
+      this.client.fetchCurrentUser()
```

Comment:
Here's the followup to my push back. We're updating the user using GraphQL, we should just query for their currencyPreference using GraphQL. The only info we want is the currency but we're pulling down the whole user object from REST. Then we won't need to update the user every time we look at a project.